### PR TITLE
Move to PSR7 only 

### PR DESCRIPTION
**Pull request by Nyholm**
Sorry for this major PR. It removes all deprecated code and it removes our implementation of HTTP request and responses.

### CODE REVIEW

File: lib/Buzz/Browser.php [hunk 1]
```diff
@@ -204,20 +120,17 @@ public function submitForm($url, array $fields, $method = RequestInterface::METH
...
     }
 
     /**
      * Send a PSR7 request.
-     *
-     * @param Psr7RequestInterface $request
-     * @return Psr7ResponseInterface
      */
-    public function sendRequest(Psr7RequestInterface $request)
+    public function sendRequest(RequestInterface $request): ?ResponseInterface
```
Comment (stof) [inline lib/Buzz/Browser.php]: why is it nullable here ? This does not make it a PSR-18 implementation

File: lib/Buzz/Browser.php [hunk 2]
```diff
@@ -40,43 +29,43 @@ class Browser
     /** @var RequestInterface */
     private $lastRequest;
 
-    /** @var MessageInterface */
+    /** @var ResponseInterface */
     private $lastResponse;
 
-    public function __construct(ClientInterface $client = null, FactoryInterface $factory = null)
+    public function __construct($client = null)
```
Comment (stof) [inline lib/Buzz/Browser.php]: accepting anything ? really

File: lib/Buzz/Browser.php [hunk 3]
```diff
@@ -289,7 +202,7 @@ public function getLastResponse()
         return $this->lastResponse;
     }
 
-    public function setClient(ClientInterface $client)
+    public function setClient($client)
```
Comment (stof) [inline lib/Buzz/Browser.php]: should it actually be mutable ?

File: lib/Buzz/Client/BatchClientInterface.php [hunk 4]
```diff
@@ -1,15 +1,17 @@
 <?php
+declare(strict_types=1);
 
 namespace Buzz\Client;
 
 use Buzz\Exception\ClientException;
+use Psr\Http\Client\ClientInterface;
```
Comment (stof) [inline lib/Buzz/Client/BatchClientInterface.php]: unused

File: composer.json [hunk 5]
```diff
@@ -13,12 +13,14 @@
         }
     ],
     "require": {
-        "php": "^5.4 || ^7.0",
-        "guzzlehttp/psr7": "^1.4"
+        "php": "^7.1",
+        "psr/http-client": "dev-master",
```
[the remark this answers is not in the document]
Comment (Nyholm) [reply composer.json]: Naturally. There were no package at the time. It is fixed here: https://github.com/kriswallsmith/Buzz/pull/272

File: lib/Buzz/Browser.php (hunk 2 above)
Comment (Nyholm) [reply lib/Buzz/Browser.php]: Yeah.. Awful. We should do the new BuzzClientInterface introduced here: https://github.com/kriswallsmith/Buzz/pull/270

File: lib/Buzz/Browser.php (hunk 3 above)
Comment (Nyholm) [reply lib/Buzz/Browser.php]: No. It should be immutable when we release this. See https://github.com/kriswallsmith/Buzz/issues/246

**Comment by Nyholm** [comment]
Thank you for the review. 
Im sorry these PRs are massive and big. I will try to do smaller from now on.

**Pull request merged**